File: crates/selene-extract/tests/fixtures/parity/csharp/Composite.cs
using System;
using System.Collections.Generic;

namespace App.Services;

public interface IUserRepository
{
    User FindById(string id);
}

public class UserService : IUserRepository
{
    private readonly IUserRepository _repository;

    public UserService(IUserRepository repository)
    {
        _repository = repository;
    }

    public User FindById(string id)
    {
        Console.WriteLine(id);
        return _repository.FindById(id);
    }
}

File: crates/selene-extract/tests/fixtures/parity/java/Composite.java
package com.example.app;

import java.util.List;
import static java.util.Collections.emptyList;

public interface Repository {
    List<String> findAll();
}

class InMemoryRepository implements Repository {
    private final List<String> items;

    InMemoryRepository(List<String> items) {
        this.items = items;
    }

    @Override
    public List<String> findAll() {
        if (items == null) {
            return emptyList();
        }
        return items;
    }
}
